.github/workflows/osx.yml

```diff
@@ -0,0 +1,87 @@
+name: github-OSX
+
+on: [push, pull_request]
+#on:
+#  pull_request:
+#    branches:
+#      - master
+#      - develop
+
+jobs:
+  osxci:
+    name: osx-ci
+    runs-on: [macos-latest]
+
+    strategy:
+      matrix:
+        include:
+          - backend: "SERIAL"
+            cmake_build_type: "RelWithDebInfo"
+          - backend: "PTHREAD"
```

Comment:
@e10harvey I think this should be THREADS now following the deprecation of PTHREAD in kokkos https://github.com/kokkos/kokkos/pull/4619